src/Composer/Satis/Command/BuildCommand.php

```diff
@@ -145,6 +146,20 @@ protected function execute(InputInterface $input, OutputInterface $output)
             throw new \InvalidArgumentException('The output dir must be specified as second argument or be configured inside '.$input->getArgument('file'));
         }
 
+        if (($singleRepositoryUrl = $input->getOption('repository-url')) !== null) {
+            $singleRepositoryConfig = null;
+            foreach ($config['repositories'] as $r) {
```

Comment:
you cannot assume that this array key exists. (nor that ``url`` exists in it) as ``$config`` is just the json-decoded content of the config file, and so you have no idea about what it contains or what is not in it.